discovery-infra/test_infra/helper_classes/cluster.py

```diff
@@ -606,8 +607,9 @@ def wait_for_install(self, timeout=consts.CLUSTER_INSTALLATION_TIMEOUT):
         )
 
     @JunitTestCase()
-    def prepare_for_installation(self, nodes: Nodes, static_network_config=None, **kwargs):
+    def prepare_for_installation(self, nodes: Nodes = None, static_network_config=None, **kwargs):
         self.update_config(**kwargs)
+        nodes = nodes or self.nodes
```

Comment:
Please describe that this exists for backward compatibility. Just because I don't see any other reason to keep it.